src/dft/backends/mklcpu/commit.cpp

```diff
@@ -0,0 +1,141 @@
+    commit_derived_impl(sycl::queue queue, const detail::dft_values<prec, dom>& config_values)
+            : detail::commit_impl(queue, backend::mklcpu) {
+        DFT_ERROR status = DFT_NOTSET;
+        if (config_values.dimensions.size() == 1) {
+            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom), 1,
+                                          config_values.dimensions[0]);
+        }
+        else {
+            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom),
+                                          config_values.dimensions.size(),
+                                          config_values.dimensions.data());
+        }
+        if (status != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
+                                         "DftiCreateDescriptor failed");
+        }
+
+        set_value(handle, config_values);
+
+        status = DftiCommitDescriptor(handle);
+        if (status != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
+                                         "DftiCommitDescriptor failed");
+        }
+    }
+
+    virtual void* get_handle() noexcept override {
+        return handle;
+    }
+
+    virtual ~commit_derived_impl() override {
+        DftiFreeDescriptor((DFTI_DESCRIPTOR_HANDLE*)&handle);
+    }
+
+private:
+    DFTI_DESCRIPTOR_HANDLE handle = nullptr;
+
+    constexpr DFTI_CONFIG_VALUE get_domain(domain d) {
+        if (d == domain::COMPLEX) {
+            return DFTI_COMPLEX;
+        }
+        else {
+            return DFTI_REAL;
+        }
+    }
+
+    constexpr DFTI_CONFIG_VALUE get_precision(precision p) {
+        if (p == precision::SINGLE) {
+            return DFTI_SINGLE;
+        }
+        else {
+            return DFTI_DOUBLE;
+        }
+    }
+
+    template <typename... Args>
+    void set_value_item(DFTI_DESCRIPTOR_HANDLE hand, enum DFTI_CONFIG_PARAM name, Args... args) {
+        if (auto ret = DftiSetValue(hand, name, args...); ret != DFTI_NO_ERROR) {
+            throw oneapi::mkl::exception(
+                "dft/backends/mklcpu", "set_value_item",
+                "name: " + std::to_string(name) + " error: " + std::to_string(ret));
+        }
+    }
+
+    void set_value(DFTI_DESCRIPTOR_HANDLE& descHandle,
+                   const detail::dft_values<prec, dom>& config) {
+        set_value_item(descHandle, DFTI_INPUT_STRIDES, config.input_strides.data());
+        set_value_item(descHandle, DFTI_OUTPUT_STRIDES, config.output_strides.data());
+        set_value_item(descHandle, DFTI_BACKWARD_SCALE, config.bwd_scale);
+        set_value_item(descHandle, DFTI_FORWARD_SCALE, config.fwd_scale);
+        set_value_item(descHandle, DFTI_NUMBER_OF_TRANSFORMS, config.number_of_transforms);
+        set_value_item(descHandle, DFTI_INPUT_DISTANCE, config.fwd_dist);
+        set_value_item(descHandle, DFTI_OUTPUT_DISTANCE, config.bwd_dist);
```

Comment:
@anantsrivastava30, @t4c1, @Rbiessy, @hjabird @lhuot: I know this PR was merged already but what motivated this usage of `DFTI_INPUT_DISTANCE` and `DFTI_OUTUT_DISTANCE` via dpc++ APIs? Those parameters are not listed in [the specs](https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/dft/enums_and_config_params.html#onemkl-dft-enums) and we explicitly recommend to use `FWD_DISTANCE` and `BWD_DISTANCE` for the usage of the closed-source oneMKL via DPC++ APIs [here](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2023-0/descriptor-precision-domain-set-value.html).

Reply:
Hey @raphael-egan the usage of these are mostly to conform to the classical API usage, that still uses these parameters, however I see you point and we can mask them in the cpu backend [pr#288](https://github.com/oneapi-src/oneMKL/pull/288), so that on the user end we still use fwd/bwd distance and later on in this part of the code change these to input/output distance, lets move this discussion to https://github.com/oneapi-src/oneMKL/pull/288

Reply:
Yes, it makes sense to move to the C++ API in the CPU MR. The initial work used the C API for the CPU backend and we didn't update this to avoid conflicts. Thanks.